images/metering-ansible-operator/roles/meteringconfig/tasks/configure_hive_metastore.yml

```diff
@@ -20,6 +20,31 @@
         message: "Configuring hive metastore"
         lastTransitionTime: "{{ now(utc=False).isoformat() + 'Z' }}"
 
+  - name: Determine the Hive underlying database
+    block:
+    - name: Dump out the URL for debugging
+      debug:
+        var: _meteringconfig_hive_metastore_db_url
+
+    - name: Split the hive.spec.config.db.url into an array
+      set_fact:
+        _url_array: "{{ _meteringconfig_hive_metastore_db_url.split(':') }}"
```

Comment:
The [`urlsplit` Ansible filter](https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#splitting-urls) wasn't able to parse JDBC URLs as I originally expected. The `scheme` comes back as `jdbc` instead of something like `postgresql` and everything past the `jdbc://...` was put in a `path` variable, which doesn't move the needle any more than using bash to do this filter for us.

In the end, I settled on using the `split` method and splitting that URL string by the `:` delimiter and validating those inputs so we have a somewhat consistent representation of what a JDBC URL should look like.